### PR TITLE
Bookinfo gherkin annotations for upcoming hook

### PR DESCRIPTION
**Pull request by matejnesuta**
This is related to the OSSM-4049 ticket. 

In order to make more robust setup, we want Cypress to detect if certain demoapps are installed before related tests. For example, if the test uses bookinfo, then all of the 6 bookinfo pods and traffic generator must be installed. If they are not or if they are malformed at some way (for example missing pod), then Cypress installs the specific demo app via the hack scripts. 

To test this:
- Deploy Kiali without the bookinfo demo app
- Run any Cypres feature file, which has the `@bookino-app` tag.
- Bookinfo demo app should be installed prior to the first test and all tests related to bookinfo from that FeatureFile should pass
- This should work both for the OCP and non-OCP.

(Drafting this with bookinfo app only. I will turn it into a regular PR once I add other demos as well.

### CODE REVIEW

**Review by jshaughn — COMMENTED** [review]
Hi @matejnesuta , This looks good.  Would it make sense to make this something that is done only once for a feature file?  Does adding the annotation on every Scenario slow things down?

**Comment by ScriptingShrimp** [comment]
> done only once for a feature file

the hooks for a "feature file" are in `Scenario:` paragraph of feature file. Definitely we can put some of the feature specific codes inhere. We decided to use conditional hook in this case because i.e. bookinfo is not related to the feature, its just a test precondition shared across multiple feature files.

  

> Does adding the annotation on every Scenario slow things down?

Definitively. But the `cy.exec` is executed outside of web browser context, so the slowdown should not be significant (I believe)

File: frontend/cypress/integration/common/hooks.ts [hunk 1]
```diff
@@ -9,3 +9,39 @@ Before({tags: '@gateway-api'}, async function () {
...
+                    `kubectl wait --for=condition=Ready pods --all -n bookinfo --timeout 60s || true;` +
+                    `kubectl wait --for=condition=Ready pods --all -n bookinfo --timeout 60s;` +
+                    `sleep 80;`,{timeout:400*1000});
+              });
+            })
+          })
+        }
+        else{
+          cy.exec('../hack/istio/install-bookinfo-demo.sh --delete-bookinfo true -c kubectl').then(()=>{
+            cy.exec('../hack/istio/install-bookinfo-demo.sh -c kubectl -tg -in istio-system -a amd64');
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/hooks.ts:41]: As we discussed, refactoring of shell scripts will be better approach.

**Review by ScriptingShrimp — CHANGES_REQUESTED** [review]
we can merge ` @bookinfo-app` annotations in feature files. 

lets refactor bookinfo installation in another PR if you want

**Comment by matejnesuta** [comment]
I commented out the hook for now, so the annotations can be merged

File: frontend/cypress/integration/common/hooks.ts [hunk 2]
```diff
@@ -9,3 +9,41 @@ Before({tags: '@gateway-api'}, async function () {
     }
   })
 });
+
+// Will be refactored soon
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/hooks.ts:13]: Please roll back the file to its original state. It's not a good practice to commit TODOs and unused code.

**Review by ScriptingShrimp — CHANGES_REQUESTED** [review]
Just remove the comments from `hooks.ts` and LGTM!
Thank you

**Comment by matejnesuta** [comment]
I removed the comments as you requested

**Pull request merged**